rpcs3/Emu/Cell/Modules/cellMic.cpp

```diff
@@ -258,6 +258,28 @@ inline void microphone_device::variable_byteswap(const void* src, void* dst)
 	}
 }
 
+inline u32 microphone_device::convert_16_bit_pcm_to_float(const std::vector<u8>& buffer, usz num_bytes)
+{
+	constexpr usz float_buf_size = 2 * inbuf_size;
+	static_assert((float_buf_size % sizeof(u32)) == 0);
+
+	float_buf.resize(float_buf_size, 0);
+	ensure(num_bytes * 2 <= float_buf.size());
+
+	be_t<f32>* dst = reinterpret_cast<be_t<f32>*>(float_buf.data());
+
+	for (usz i = 0; i < num_bytes; i += 2)
+	{
+		const s16 sample = static_cast<s16>((buffer[i] << 8) | buffer[i + 1]);
```

Comment:
Isn't this just extracting big endian s16? (correct me if I'm wrong)
So you can probably just use the same trick again:

Before the loop:
`const be_t<s16>* src = reinterpret_cast<const be_t<s16>*>(buffer.data());`

And then do:
`for (usz i = 0; i < num_bytes / 2; i++)`

In the loop:
`const be_t<s16> sample = src++;`